fix(smoke_test): fix smoke test failing with big nodes

TauTheLepton opened the pull request:
Related to this issue autowarefoundation/autoware.universe#2684

## Description

Fixed smoke test to not fail while testing big (taking lots of time to start) nodes.
Added timeout after which smoke test is considered failed (2 seconds).
Added parameters file for testing purposes.

## Tests performed

Running `colcon test` on `trajectory_follower_node` now passes.
When I disabled `trajectory_follower_node` the test timed out after 2 seconds like it should.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

TakaHoribe commented on control/trajectory_follower_node/param/test.param.yaml:
```diff
@@ -0,0 +1,12 @@
+    longitudinal_controller_mode: "pid"
+
+    # steer offset
+    steering_offset:
```
These parameters must be in the lateral_controller param as in autoware_launch.
https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/mpc.param.yaml#L66-L72

Would you move them into [here](https://github.com/autowarefoundation/autoware.universe/blob/9983ead76b3fc164f23aab6d8fe1d27519f1770c/control/trajectory_follower_node/param/lateral_controller_defaults.param.yaml)?

TauTheLepton replied:
Check my latest comment, whether it resolves your issue.

TakaHoribe commented on control/trajectory_follower_node/param/test.param.yaml:
```diff
@@ -0,0 +1,12 @@
+/**:
+  ros__parameters:
+    lateral_controller_mode: "mpc"
```
Since these params are only used for the test, would you move these under `test` directory? 
https://github.com/autowarefoundation/autoware.universe/tree/9983ead76b3fc164f23aab6d8fe1d27519f1770c/control/trajectory_follower_node/test

Or, ideally, we should test for all options (not only for the `mpc` but also `pure_pursuit`), so I guess these params should not be in the param file, but could be defined in the test directly.

TauTheLepton replied:
Check my latest comment, whether it resolves your issue.

TakaHoribe replied:
thank you!

TakaHoribe reviewed (commented):
@TauTheLepton Thank you for the nice work! Would you check my comments about the parameter definitions?

TauTheLepton commented:
@TakaHoribe I moved the parameters as you wanted.
I also extended smoke test functionality to support multiple smoke tests per node and support testing only parameters (located in `test` directory). This is, in my opinion, the best way to deal with testing multiple configurations of the same node - create multiple testing parameter files and launch different smoke tests with every such file.
I added this functionality to `trajectory_follower_node`, `pure_pursuit` controller fails the test which is expected and I already opened an issue about this (#3309).

kyoichi-sugahara commented:
@TauTheLepton 
Thank you so much for your work!
When I look at the lateral_controller param in autoware_launcher, param file stracture as below,
```
~/hoge/hoge/autoware_launch/config/control/trajectory_follower/lateral/
├── mpc.param.yaml
└── pure_pursuit.param.yaml
```
So it's better to move/keep parameters related pure_pusit in the [pure_pursuit.param.yaml](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/pure_pursuit.param.yaml).

If my understand is not wrong, Mr. @TakaHoribe said below is
> These parameters must be in the lateral_controller param as in autoware_launch.
https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/mpc.param.yaml#L66-L72
Would you move them into [here](https://github.com/autowarefoundation/autoware.universe/blob/9983ead76b3fc164f23aab6d8fe1d27519f1770c/control/trajectory_follower_node/param/lateral_controller_defaults.param.yaml)

not move pure_pusit related param into the `[lateral_controller_defaults.param.yaml](https://github.com/autowarefoundation/autoware.universe/pull/3287/commits/14bb4ec8786e6e0b443266c732039ce078dbd239#diff-e0835a03a9b2304646decbdd8a9529d532973d1994c702eac8b1c783c1ea33a9)`

Do you have any reason move those params to `lateral_controller_defaults.param.yaml` or system limitation?

TauTheLepton commented:
@kyoichi-sugahara 

> Do you have any reason move those params to `lateral_controller_defaults.param.yaml` or system limitation?

The only reason it is this way is because the file `lateral_controller_defaults.param.yaml` already existed before I started working on this issue. I just kept the old file structure.

I have a few questions about what you would like the solution to look like

1. Move all parameters associated with mpc controller to `mpc.param.yaml` like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/mpc.param.yaml) and all parameters associated with pure pursuit controller to `pure_pursuit.param.yaml` like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/pure_pursuit.param.yaml)?
1. If so, then maybe the same should be done with the `longitudinal_controller_defaults.param.yaml` - change it to become a `pid.param.yaml` like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/longitudinal/pid.param.yaml)?
1. These controller parameter files should be located in the `param` directory or in specific sub-directories `param/lateral` and `param/longitudinal` like [here](https://github.com/autowarefoundation/autoware_launch/tree/main/autoware_launch/config/control/trajectory_follower)?
1. What to do with the controller type declaration parameters, leave them as is in the `test` directory?

kyoichi-sugahara commented:
@TauTheLepton 

I see!
>The only reason it is this way is because the file lateral_controller_defaults.param.yaml already existed before I started working on this issue. I just kept the old file structure.

basic idea is parameter yaml file structure including parameters in the file should be same with autoware_launch. 
So yes. I think so. (but file name can be defined as `test_***` so that autoware user will not get confused, if those parameters are used only for the test)
>Move all parameters associated with mpc controller to mpc.param.yaml like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/mpc.param.yaml) and all parameters associated with pure pursuit controller to pure_pursuit.param.yaml like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/lateral/pure_pursuit.param.yaml)?

Yes I think so.
>If so, then maybe the same should be done with the longitudinal_controller_defaults.param.yaml - change it to become a pid.param.yaml like [here](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/control/trajectory_follower/longitudinal/pid.param.yaml)?

It's better that having same folder structure.
These controller parameter files should be located in the param directory or in specific sub-directories param/lateral and param/longitudinal like [here](https://github.com/autowarefoundation/autoware_launch/tree/main/autoware_launch/config/control/trajectory_follower)?

If it's parameter, I think this controller type declaration param in the yaml file should be in the param dirctory.
What to do with the controller type declaration parameters, leave them as is in the test directory?

kyoichi-sugahara commented:
@TauTheLepton 
Thank you so much for your change!
From my side, the change you made is good!

kyoichi-sugahara commented:
@TakaHoribe 
Can you check this PR when you have time?

TakaHoribe reviewed (approved):
great. Thank you!
